R/human_infection.R

```diff
@@ -181,23 +182,98 @@ calculate_infections <- function(
       alpha[pev_profile]
     )
   }
-
+  
   prob <- b * (1 - prophylaxis) * (1 - vaccine_efficacy)
-  infected <- bitset_at(source_humans, bernoulli_multi_p(prob))
-
-  incidence_renderer(
-    variables$birth,
-    renderer,
-    infected,
-    source_humans,
-    prob,
-    'inc_',
-    parameters$incidence_rendering_min_ages,
-    parameters$incidence_rendering_max_ages,
-    timestep
-  )
-
-  infected
+  
+  ## All bitten humans with an infectious bite (incorporating prophylaxis and vaccines)
+  newly_infected <- bitset_at(source_humans, bernoulli_multi_p(prob))
+  
+  if(parameters$parasite == "falciparum"){
+    # Render new infections caused by bites
+    renderer$render('n_infections', newly_infected$size(), timestep)
+    incidence_renderer(
+      variables$birth,
+      renderer,
+      newly_infected,
+      source_humans,
+      prob,
+      'inc_',
+      parameters$incidence_rendering_min_ages,
+      parameters$incidence_rendering_max_ages,
+      timestep
+    )
+    total_infected <- newly_infected
+  } 
+  
+  else if(parameters$parasite == "vivax"){
+    
+    SAU_humans <- variables$state$get_index_of(c('S','A','U'))
+    new_bite_infections <- newly_infected$and(SAU_humans)
+    
+    # Render new infections caused by bites
+    renderer$render('n_new_bite_infections', new_bite_infections$size(), timestep)
```

Comment:
It's not quite the same because an individual can experience a relapse and a new infection at the same time. They aren't necessarily all important outputs (I thought I would include more so we had options), but possibly worth having a conversation about.